internal/proxy/proxy.go

```diff
@@ -282,6 +256,58 @@ func NewClient(ctx context.Context, cmd *cobra.Command, conf *Config) (*Client,
 	return &Client{mnts: mnts, cmd: cmd, dialer: d}, nil
 }
 
+// dialerOptions builds appropriate list of options from the Config
+// values for use by cloudsqlconn.NewClient()
+func dialerOptions(c *Config) ([]cloudsqlconn.Option, error) {
```

Comment:
I was trying to make the call signatures for these functions consistent: 
 ```
func (c *Config) DialerOptions() ([]cloudsqlconn.Option, error)  {}
func dialOptions(conf *Config, inst *InstanceConnConfig) []cloudsqlconn.DialOption {} 
``` 

So when Eno and I discussed the inconsistency, we landed on this:
 ```
func dialerOptions(c *Config) ([]cloudsqlconn.Option, error)  {}
func dialOptions(conf *Config, inst *InstanceConnConfig) []cloudsqlconn.DialOption {} 
``` 

The alternative to make `DialOptions` a  member functions of *Config, which seemed worse to me because
`DialOptions` is more transformation on Config and InstanceConnConfig, and less a member of Config.
```
func (c *Config) DialerOptions() ([]cloudsqlconn.Option, error)  {}
func (c *Config) DialOptions(inst *InstanceConnConfig) ([]cloudsqlconn.Option, error)  {}
```

Reply:
The difference between function vs method is largely syntactic / organizational. For me, it more sense to me to use `func (c *Config) DialerOptions()` and `func (i *instanceConnConfig) dialOptions()` (and making InstanceConnConfig contain a reference to it's parent config) because they signify a way of transforming the object to another type of data, similar to how `String()` works. It's also easier to discover (because it's attached to the object it transforms, and lands closer to it in the doc) and leaves the door open to more complex behavior (say in the future it needs to be thread safe, we can add an internal mutex more easily).

Reply:
I think both approaches are reasonable, but if there's preference for methods then let's do that.

Reply:
Ok, i'll try again with a version closer to @kurtisvg's suggestion.